x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/job/RollupIndexer.java

```diff
@@ -54,6 +55,7 @@
     private final AtomicReference<IndexerState> state;
     private final AtomicReference<Map<String, Object>> position;
     private final Executor executor;
+    protected final AtomicBoolean upgradedDocumentID;
```

Comment:
Note: an atomic is probably overkill here... only the indexer thread can update it, so we _probably_ could have used a volatile here.  But I didn't want to mess around with concurrency semantics on this bugfix so went with a simple, easily-reasoned atomic.